esp-hal/src/i2c.rs

```diff
@@ -387,6 +423,25 @@ where
             crate::interrupt::enable(T::interrupt(), handler.priority()).unwrap();
         }
     }
+
+    /// Recover from error condition
+    pub fn recover(&self) {
```

Comment:
This should probably be `&mut self` right? If it needs to be `&self` for internal lifetime reasons, we should have two methods one public and one private.

Reply:
ah - I didn't plan to make this public at all

if public it needs to be `&mut self` of course - I didn't thought this would be something we want in public API, do we?
If yes I will expose a public variant taking `&mut self`

Reply:
Let's put this back to private. In theory the driver should self recover, and if it doesn't that's a bug that users should report.